### DISCUSSION TITLE
How to test Livewire Forms

### DISCUSSION THREAD

**Discussion by cstisa** [livewire/livewire#8104] [1. Help] [answered]
Hi all,
I have difficulties to test the Form that I've created.
Especially, I don't know if I got to test my livewire form by itself or inside the component that uses it?
Moreover, if I got to test a livewire component that needs a livewire form as property, I don't know how do instantiate my livewire form and I can't find any doc about it.

My goal here is to be able to test the methods inside my livewire Form

**Comment by joshhanley** [comment, marked as answer]
@cstisa hmm yeah it seems there is no docs on testing forms.

Personally, I just test the form through the component.

```php
it('does something on a form', function() {
    Livewire::test(MyComponent::class)
        ->set('form.title', 'My Title')
        ->call('save') // The save method on the component should call save on the form so we can test it this way.
});
```

**Reply by cstisa** [reply]
That's what I thought. I ended up doing the same, testing the form from the component that uses it.

Thanks for the answer anyway!

**Comment by cstisa** [comment]
Another quick question for testing forms.

I have a Component that needs a livewire Form as a parameter like this :

```
Livewire::test(AddModelDialog::class, [
            'form' => $form,
            'formComponent' => $this->formComponent,
            'stringReplacements' => $stringReplacements,
        ])
            ->assertStatus(200);
```

The problem that I have is that I don't know how to create my livewire Form $form by itself in a test.

I tried this but it seems way harder than it should be : 

```
$mockComponent = Livewire::test(AddModelDialog::class)
            ->partialMock();
        $form = new \App\Livewire\Forms\Career($mockComponent, 'model'); // Use real component and property name
```

Is there a proper way to init a livewire Form in a test in order to pass it as a parameter for the component that needs it and that I need to test?

**Reply by joshhanley** [reply]
@cstisa hmm not sure as I've never done that. Why are you passing the form into the child component instead of letting the child component control it?
